proxylib/cassandra/cassandraparser.go

```diff
@@ -206,14 +206,14 @@ func (p *CassandraParser) OnData(reply, endStream bool, dataArray [][]byte, offs
 		}
 		cassandraParseReply(p, data[0:(cassHdrLen+requestLen)])
 
-		log.Debugf("reply, passing %d bytes", uint32(len(data)))
-		return PASS, len(data)
+		log.Debugf("reply, passing %d bytes", (cassHdrLen + requestLen))
+		return PASS, (cassHdrLen + int(requestLen))
```

Comment:
This is the key fix in the change-set.   Returning len(data) can end up passing more data than just the single request depending on data buffering, which would then leave us in the middle of a subsequent request the next time OnData was called.